admiral/pkg/clusters/serviceentry.go

```diff
@@ -774,6 +757,24 @@ func modifyServiceEntryForNewServiceOrPod(
 		for _, val := range dependents {
 			remoteRegistry.AdmiralCache.DependencyNamespaceCache.Put(val, serviceInstance[appType[sourceCluster]].Namespace, localFqdn, cnames)
 		}
+
+		if common.IsVSBasedRoutingEnabled() {
+			// Discovery phase: This is where we build a map of all the svc.cluster.local destinations
+			// for the identity's source cluster. This map will contain the RouteDestination of all svc.cluster.local
+			// endpoints.
+			destinations, err := getAllVSRouteDestinationsByCluster(
+				serviceInstance,
+				meshDeployAndRolloutPorts,
+				sourceWeightedServices[sourceCluster],
+				sourceRollouts[sourceCluster],
+				sourceDeployments[sourceCluster])
+			if err != nil {
+				ctxLogger.Errorf(common.CtxLogFormat, "getAllVSRouteDestinationsByCluster",
```

Comment:
Why are we not returning from here?

Reply:
@kpharasi appended to `modifySEerr`, so that it can be retried.